### DISCUSSION TITLE
Increase speed

### DISCUSSION THREAD

**Discussion by gzzzze** [glucauze/sd-webui-faceswaplab#149] [General]
this plugin is great and I'm quite happy with the results I'm able to obtain, however in this moment each image will take more or less 6 seconds to be generate.

I was wondering if I can increase the speed in some way, is this dependend more on CPU, GPU or?

![image](https://github.com/glucauze/sd-webui-faceswaplab/assets/150005010/d71254a4-cb1f-4742-b598-ec5f6ee15186)

**Comment by Moussack** [comment]
Is that using faceswaplab in img2img tab or in the native faceswaplab tab??
try swaping in faceswaplab tab batch proccess, it's faster than swaping with fsl through img2img or txt2img tab, because it bypass the a1111 pipeline.

Or even better, enable GPU Cuda in the option so FSL can use your GPU instead CPU for even faster swapping. And don't forget to completely restart the A1111 by closing the terminal after your swapping session, because the swapper model will stay in vram and not released (your vram will always full and if you use A1111 to generate txt2img or img2img later without restarting the UI, the image generation will be very slow ). You can release the vram by completely restart the A1111.

**Comment by gzzzze** [comment]
I'm using img2img because i need to batch process thousand of pics, can i do the same directly in the fs tab?

**Reply by Moussack** [reply]
Yes, you can do batch process directly in fs tab as well like in img2img, i do it all the time for faster faceswap thousands of picture frames, you can do it in "Batch process" tab.

![chrome_59fqXBH7pO](https://github.com/glucauze/sd-webui-faceswaplab/assets/69501089/6a82e701-fed1-41ea-90f6-b73f1acda6af)


img2img is slower bcuz it calls img2img function first and then faceswaplab function. 
In fs tab it only calls the faceswaplab function, hence faster faceswap.

In my old setup with img2img, it takes 10 to 16 sec to faceswap an image.
in fs tab it takes only 4 to 5 sec with codeformer and less than 1 sec without codeformer to faceswap. 
That all with "use gpu" option and with my ancient PC setup (Nvidia GTX 970, 16GB RAM, and i5-4690K). 
Your system probably can do faster.

**Reply by gzzzze** [reply]
thanks! the issue in that tab is that I cannot select a folder for source images but instead I need to drop all them and when they are thousands it lock the system. instead with the folder selection it work smoothly

**Comment by gzzzze** [comment]
or better what I mean is that in img2img I can simply select a directory for batch process while in the fs tab I need to upload each single image
